lib/iris/tests/__init__.py

```diff
@@ -774,6 +775,82 @@ def assertGribMessageContents(self, filename, contents):
                 section, key, val = element
                 self.assertEqual(message.sections[section][key], val)
 
+    def assertGribMessageDifference(self, filename1, filename2, diffs,
+                                    skip_keys=None):
+        """
+        Evaluate that the two messages only differ in the ways specified.
+
+        * filename[0|1] (string)
+            The path on disk of existing GRIB files
+
+        * diffs
+            An dictionary of GRIB message keys and expected diff values:
+            {key: (m1val, m2val),...} .
+
+        * skip_keys
+            An iterable of key names to ignore during comparison.
+
+        """
+        if skip_keys is None:
+            skip_keys = []
+        messages1 = _GribMessage.messages_from_filename(filename1)
+        messages2 = _GribMessage.messages_from_filename(filename2)
+        for m1, m2 in zip_longest(messages1, messages2):
```

Comment:
I think catching and raising if the two files produce different number of messages would be sensible.